torchvision/models/detection/retinanet.py

```diff
@@ -487,8 +487,8 @@ def forward(self, images, targets=None):
                 like `scores`, `labels` and `mask` (for Mask R-CNN models).
 
         """
-        if self.training and targets is None:
-            raise ValueError("In training mode, targets should be passed")
+        if self.training:
+            assert targets is not None
```

Comment:
The if statement 2 lines below does the exact same check. Shall we drop this completely? Also what about the ValueErrors in the rest of this method?

Reply:
You are right, I missed those because I went though all asserts that were replaced with exceptions. Those were there already as exceptions, but in the spirit of this PR should I change those too?